back/lib/participation_method/base.rb

```diff
@@ -118,11 +118,7 @@ def supports_permitted_by_everyone?
       false
     end
 
-    def supports_posting_inputs?
-      false
-    end
-
-    def supports_public_visibility?
+    def supports_posting?
```

Comment:
I think what I meant with "posts" is slightly different from "input based", because I wanted to give a name to the concept of a submission that is also publicly visible. So survey responses are not posts, but they are input based. Today we already a `Post` concern that is used for ideas and proposals, so I thought it would be useful to reuse this concept. I was doubting about bringing it up during our standup or even the CDM, because I'm indeed using terminology that we didn't really explicitly agree on (but the codebase tends to implicitly agree).

- Input: An idea, proposal or survey response.
- Post: An input that is publicly visible. Ideas and proposals are posts, survey responses are not. I could also imagine having a `Post` model at some point (similar to the `AdminPublication` we have today), so that the same input could be posted in multiple places. But by moving the concept of a "post" to the participation method (`supports_posting?`) we could potentially even decide to make some survey responses visible to the public.
- Submission: An input that is "finalized" in the sense that it is not / no longer a draft.
- Published: I felt like maybe we could get rid of this concept in some parts and use "posted" instead. For example, we could have a `publication_status` with as possible values `draft`, `submitted` and `published`.

To me it felt like this would make it easier to understand the publication statuses and the different types of input. Do you think there is any problems with this reasoning? Is there anything that could be confusing around this?

Reply:
In the permissions, we have the actions `posting_idea` and `posting_initiative`, but I think these are also used for survey responses. So in this case, I would suggest to rename the action to `submitting`.

Reply:
Alright, thanks for the explanation, I think that I understand it now.

I think what maybe confused me most, is that posting is just a regular verb, where the verb has little to do with it, in this case. It's not about the act of posting. It's about
(a) this method storing its inputs in the ideas table and
(b) this being publicly visible

Maybe it might even be easier to keep these methods separete, and just call them both? It would be a lot easier to understand for the reader, even if it's a bit more verbove. I also wonder how often we actually need to check (a), give that the code where we call this is often probably already only running for input-based methods?

If you decide to leave it as is, could you add some proper comments to the method implementation (just in the `Base` class is maybe ok) to explain what these methods mean?

Reply:
Indeed it's not so much related to "posting" as a verb, but I thought perhaps more like "post" as a noun (an input that you want to be visible to the public).

And indeed we wouldn't have to check a). In fact, what I did is simply rename `supports_public_visibility?` to `supports_posting?`. So I think your suggestion comes down to reverting this renaming? This is definitely something I can do.

Reply:
Yes, I think it actually does, and I think it would be better. I understand why you changed it, but from the perspective of someone new to the codebase, it would be very confusing.